File: example/java/jimmer-cloud/author-service/src/main/java/org/babyfish/jimmer/example/cloud/author/cfg/DatabaseInitializer.java
package org.babyfish.jimmer.example.cloud.author.cfg;

import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.runtime.Initializer;
import org.springframework.stereotype.Component;

import javax.sql.DataSource;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.sql.Connection;

@Component
public class DatabaseInitializer implements Initializer {

    private final DataSource dataSource;

    public DatabaseInitializer(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    @Override
    public void initialize(JSqlClient sqlClient) throws Exception {
        try (Connection con = dataSource.getConnection()) {
            InputStream inputStream = DatabaseInitializer.class
                    .getClassLoader()
                    .getResourceAsStream("author.sql");
            if (inputStream == null) {
                throw new RuntimeException("no `author.sql`");
            }
            try (Reader reader = new InputStreamReader(inputStream)) {
                char[] buf = new char[1024];
                StringBuilder sqlBuilder = new StringBuilder();
                while (true) {
                    int len = reader.read(buf);
                    if (len == -1) {
                        break;
                    }
                    sqlBuilder.append(buf, 0, len);
                }
                con.createStatement().execute(sqlBuilder.toString());
            }
        }
    }
}

File: example/kotlin/jimmer-cloud-kt/author-service/src/main/kotlin/org/babyfish/jimmer/example/cloud/kt/author/cfg/DatabaseInitializer.kt
package org.babyfish.jimmer.example.cloud.kt.author.cfg

import org.babyfish.jimmer.sql.kt.KSqlClient
import org.babyfish.jimmer.sql.kt.cfg.KInitializer
import org.springframework.stereotype.Component
import java.io.InputStreamReader
import javax.sql.DataSource

@Component
class DatabaseInitializer(private val dataSource: DataSource) : KInitializer {

    override fun initialize(dsl: KSqlClient) {
        dataSource.connection.use { con ->
            val inputStream = JimmerConfig::class.java
                .classLoader
                .getResourceAsStream("author.sql") ?: throw RuntimeException("no `author.sql`")
            InputStreamReader(inputStream).use { reader ->
                con.createStatement().execute(reader.readText())
            }
        }
    }
}